src/features/loupe/ImageReviewToolbar.jsx

```diff
@@ -274,6 +294,35 @@ const ImageReviewToolbar = ({
 
           <Separator />
 
+          {/* Comments */}
+          <Tooltip>
+            <PopoverRoot open={isCommentsPopoverOpen}>
+              <TooltipTrigger asChild disabled={!hasRole(userRoles, READ_COMMENT_ROLES) || !hasRole(userRoles, WRITE_COMMENT_ROLES)}>
+                <PopoverTrigger asChild onClick={() => setIsCommentsPopoverOpen(true)}>
+                  <ToolbarIconButton>
+                    <ChatBubbleIcon />
+                  </ToolbarIconButton>
+                </PopoverTrigger>
+              </TooltipTrigger>
+              <TooltipContent side="top" sideOffset={5}>
+                Add comments
+                <TooltipArrow />
+              </TooltipContent>
+              <PopoverPortal>
+                <StyledPopoverContent side="top" sideOffset={25} onPointerDownOutside={() => {}}>
```

Comment:
**TODO for self**: Seems like I forgot to remove the handler after emptying it.  I should remove `onPointerDownOutside` because it's doing nothing right now 

When the child dropdown menu (three dots edit / delete) is open, the popover's `onPointerDownOutside` gets triggered by any click outside the bounds of the dropdown menu (not the popover).  

https://github.com/user-attachments/assets/004bb0d9-7303-4dc9-b516-42bdbde37798


I tried also making both controlled components and ignoring the `onPointerDownOutside` if the child dropdown but it caused the action menu to take two clicks to open.  Here's a really simplified explanation of what I was trying:
```tsx
const Toolbar = () => {
  const [commentsOpen, setCommentsOpen] = useState(false)
  const [childOpen, setChildOpen] = useState(false)

  return (
    <Toolbar>
      <CommentsPopover>
        <Comment onOpen={() => setChildOpen(true)}  onClose={() => setChildOpen(false)} />
        <Comment onOpen={() => setChildOpen(true)}  onClose={() => setChildOpen(false)} />
        ...
      </CommentsPopover>
    </Toolbar>
  )
}

const Comment = ({ onOpen, onClose }) => {
  const [selfIsOpen, setSelfIsOpen] = useState(false)
  const handleClickToOpen = () => {
    // Setting the parent's state here causes the action menu to take two clicks to open
    onOpen()
    setSelfIsOpen(false)
  }
  return (
    <CommentContent />
  )
}
```

Reply:
That's super odd... I think we're experiencing different behavior on our browsers. I'm on a Mac and tried the code below with both Chrome and Firefox and you'll see in the video that when the dropdown is open and I click the popover behind it the `onPointerDownOutside` event does not fire. 

```javascript
        ...
        <StyledPopoverContent
          side="top"
          sideOffset={25}
          onPointerDownOutside={() => {
            console.log('onPointerDownOutside');
            setIsCommentsPopoverOpen(false);
          }}
        >
          <CommentsPopover
            onClose={() => setIsCommentsPopoverOpen(false)}
            comments={image.comments}
            imageId={image._id}
          />
          <StyledPopoverArrow />
        </StyledPopoverContent>
        ...
```
https://github.com/user-attachments/assets/50526742-d79b-40db-87f0-f1e2cdd01f01

Reply:
Weird.  I wonder if it's because I'm using Firefox Nightly so it's one of the instabilities that comes along with it.  I read-added some code to allow the popover to be closed from a click outside if the child action menu isn't open.  Seems like the popover still loses track of its own size once the child menu opens so it has that slightly weird behavior you mentioned.  

> the only friction is that if the dropdown is open and you click both outside the dropdown and the popover, on the first click just the dropdown closes, so it takes two clicks for the popover itself to close.

Frustrating but it's better than nothing.  Let me know if these changes are working well for you.  I like the addition of the comments count btw.

Reply:
This looks good to me! I'm going to merge and deploy now :).